lib/ansible/modules/cloud/kubevirt/kubevirt_scale_vmirs.py

```diff
@@ -193,56 +191,146 @@ def execute_module(self):
+        except KubernetesException as exc:
+            self.fail_json(msg='Failed to retrieve requested object: {0}'.format(exc.message),
+                           error=exc.value.get('status'))
+
+        if hasattr(existing.spec, 'replicas'):
+            existing_count = existing.spec.replicas
+
+        if existing_count is None:
+            self.fail_json(msg='Failed to retrieve the available count for the requested object.')
+
+        if resource_version and resource_version != existing.metadata.resourceVersion:
+            self.exit_json(**return_attributes)
+
+        if current_replicas is not None and existing_count != current_replicas:
+            self.exit_json(**return_attributes)
+
+        if existing_count != replicas:
+            return_attributes['changed'] = True
+            if not self.check_mode:
+                k8s_obj = self.scale(resource, existing, replicas, wait, wait_time)
+                return_attributes['result'] = k8s_obj.to_dict()
+
+        self.exit_json(**return_attributes)
+
+    @property
+    def argspec(self):
+        args = copy.deepcopy(COMMON_ARG_SPEC)
+        args.pop('state')
+        args.pop('force')
+        args.update(AUTH_ARG_SPEC)
+        args.update(VMIR_ARG_SPEC)
+        return args
+
+    def scale(self, resource, existing_object, replicas, wait, wait_time):
+        name = existing_object.metadata.name
+        namespace = existing_object.metadata.namespace
+
+        scale_obj = {'metadata': {'name': name, 'namespace': namespace}, 'spec': {'replicas': replicas}}
+
+        return_obj = None
+        stream = None
+
+        if wait:
+            w, stream = self._create_stream(resource, namespace, wait_time)
+
+        try:
+            self.patch_resource(resource, scale_obj, existing_object, name,
+                                namespace, merge_type='merge')
+        except Exception as exc:
+            self.fail_json(
+                msg="Scale request failed: {0}".format(exc.message)
             )
-        except (IOError, ConfigException):
-            raise
-
-    def __configure_by_params(self):
-        kube_config.load_kube_config()
-        config = kubernetes.client.Configuration()
-        auth_args = AUTH_ARG_SPEC.keys()
-
-        for key, value in iteritems(self.params):
-            if key in auth_args and value is not None:
-                if key == 'api_key':
-                    setattr(
-                        config,
-                        key, {'authorization': "Bearer {0}".format(value)})
-                else:
-                    setattr(config, key, value)
-
-        if not self.params.get('verify_ssl'):
-            setattr(config, 'verify_ssl', False)
-
-        return kubernetes.client.ApiClient(configuration=config)
+
+        if wait and stream is not None:
+            return_obj = self._read_stream(resource, w, stream, name, replicas)
+
+        if not return_obj:
+            return_obj = self._wait_for_response(name, namespace)
```

Comment:
Why do we need to wait if we can use events to understand whether the cluster got to required state?

Reply:
In case the `wait=false`

Reply:
My point was that we use `time.sleep(2)` and we poll. I thought it would be better to rely our logic on   a `watch` instead.